docs/modules/ROOT/pages/how-to-guides/Troubleshooting.adoc

```diff
@@ -0,0 +1,120 @@
+
+
+
+[[insufficient-resources]]
+== Insufficient Resources
+
+=== Symptoms
+Your application fails to start, or you receive errors related to resource limits.
+
+=== Troubleshooting Steps
+1. **Check Quotas**: Verify if you've hit your resource quotas for CPU, memory, or storage in your Payara Cloud Dashboard.
+2. **Scale Resources**: If needed, adjust the allocated resources for your application.
+3. **Review Logs**: Look for any messages about resource shortages in the logs.
+
+[[conflicting-ports]]
+== Conflicting Ports
+
+=== Symptoms
+The application doesn't start, and you see errors related to port allocation.
+
+=== Troubleshooting Steps
+1. **Check Configuration**: Verify that your application is configured to use ports that are available and not already in use by another application.
+2. **Review Logs**: Look for errors related to port conflicts in the logs.
+
+[[timeout-errors]]
+== Timeout Errors
+
+=== Symptoms
+The application takes too long to start, resulting in a timeout error.
+
+=== Troubleshooting Steps
+1. **Check Logs**: Review logs for any long-running processes or hang-ups during startup.
+2. **Adjust Timeout Settings**: Increase the timeout settings if your application requires more time to start.
+
+[[wrong-java-jakarta-runtime]]
+== Incompatible Runtime Environment
+
+=== Symptoms
+Your application fails to run and logs indicate incompatible Java/Jakarta EE versions or missing dependencies.
+
+=== Troubleshooting Steps
+1. **Check Runtime Version**: Make sure that your application is compatible with the Java (supported versions are 11, 17) andj Jakarta EE (supported versions are , 10) versions running in Payara Cloud.
+2. **Review Dependencies**: Ensure all required dependencies are included in your deployment package.
+
+[[database-not-connecting]]
+== Database Connection Issues
+
+=== Symptoms
+Your application cannot connect to the database, resulting in application errors.
+
+=== Troubleshooting Steps
+1. **Verify Connection Strings**: Make sure your database connection strings are correctly configured on the Payara Cloud Dashboard.
+2. **Check Network Rules**: Ensure that Payara Cloud has access to your database, especially if it's hosted outside of Azure.
+3. **Review Logs**: Inspect logs for database-related errors or warnings.
+
+[[deployment-script-errors]]
+== Deployment Script Errors
+
+=== Symptoms
+Automated deployment fails due to script errors.
+
+=== Troubleshooting Steps
+1. **Review Scripts**: Check your deployment scripts for any errors or misconfigurations.
+2. **Check Exit Codes**: Look at the exit codes in the logs to identify which part of the script failed.
+
+
+
+== General Troubleshooting Tips
+
+1. **Review Logs**: Payara Cloud provides extensive logging. Always make it a point to check the logs for any abnormalities or error messages.
+2. **Contact Support**: If all else fails, it may be beneficial to contact Payara support for more specialized assistance.
```

Comment:
```suggestion
2. **Obtain heap dumps or thread dumps**: In order to diagnose what your application is doing having either of these tools might help.
```
Support is only offered at enterprise tier, so it is not really an option for mere mortals. But we can suggest using the forum, maybe.